src/dbnode/persist/fs/read.go

```diff
@@ -50,6 +50,8 @@ var (
 
 	// errReadNotExpectedSize returned when the size of the next read does not match size specified by the index
 	errReadNotExpectedSize = errors.New("next read not expected size")
+
+	errUnexpectedSortByOffset = errors.New("should not sort index by offsets when doing reads sorted by Id")
```

Comment:
nit: I don't think this can ever hit?

Reply:
I remember adding this check to ensure `readIndexAndSortByOffsetAsc` is not called from `Open` when it is not needed. It is expensive operation, would be effectively a noop with `OrderedByIndex=true`, and I cannot check whether it was called or not from tests.
OTOH the choice between `readInIndexedOrder` vs `readInStoredOrder` is really explicit and colocated, I feel that such checks there would be excessive.